examples/tree_2d_dgsem/elixir_kpp.jl

```diff
@@ -0,0 +1,146 @@
+using Trixi
+using OrdinaryDiffEq
+using Plots
+using LinearAlgebra
+
+###############################################################################
+# Definition of the 2D scalar "KPP equation"
+#
+# See: Kurganov, A., Petrova, G., and Popov, B. (2007).
+# Adaptive Semidiscrete Central-Upwind Schemes for Nonconvex Hyperbolic Conservation Laws,
+# SIAM Journal on Scientific Computing, 29(6), 2381--2401
+# DOI: https://doi.org/10.1137/040614189
+
+struct KPPEquation2D <: Trixi.AbstractEquations{2, 1} end
+
+# The KPP flux is F(u) = (sin(u), cos(u))
+@inline function Trixi.flux(u, orientation::Integer, ::KPPEquation2D)
+    if orientation == 1
+        return SVector(sin(u[1]))
+    else
+        return SVector(cos(u[1]))
+    end
+end
+
+# Since the KPP problem is a scalar equation, the entropy-conservative flux is uniquely determined
+@inline function Trixi.flux_ec(u_ll, u_rr, orientation::Integer, ::KPPEquation2D)
+    # The tolerance of 1e-12 is based on experience and somewhat arbitrarily chosen
+    if abs(u_ll[1] - u_rr[1]) < 1e-12
+        return 0.5 * (flux(u_ll, orientation, KPPEquation2D()) + flux(u_rr, orientation, KPPEquation2D()))
+    else
+        factor = 1.0 / (u_rr[1] - u_ll[1])
+        if orientation == 1
+            return SVector(factor*(-cos(u_rr[1]) + cos(u_ll[1])))
+        else
+            return SVector(factor*(sin(u_rr[1]) - sin(u_ll[1])))
+        end
+    end
+end
+
+# Wavespeeds
+@inline wavespeed(::KPPEquation2D) = 1.0
+@inline Trixi.max_abs_speeds(u, equation::KPPEquation2D) = (wavespeed(equation), wavespeed(equation))
+@inline Trixi.max_abs_speed_naive(u_ll, u_rr, orientation::Integer, equation::KPPEquation2D) = wavespeed(equation)
+@inline Trixi.max_abs_speed_naive(u_ll, u_rr, normal_direction::AbstractVector, equation::KPPEquation2D) = wavespeed(equation) * norm(normal_direction)
+
+# Compute entropy: we use the square entropy
+@inline Trixi.entropy(u::Real, ::KPPEquation2D) = 0.5 * u^2
+@inline Trixi.entropy(u, ::KPPEquation2D) = entropy(u[1], equation)
+
+# Convert between conservative, primitive, and entropy variables. The conserved quantity "u" is also
+# considered the "primitive variable". Since we use the square entropy, "u" is also the entropy
+# variable.
+@inline Trixi.cons2prim(u, ::KPPEquation2D) = u
+@inline Trixi.cons2entropy(u, ::KPPEquation2D) = u
+
+Trixi.varnames(::Any, ::KPPEquation2D) = ("u",)
```

Comment:
When I wanted to have the save solution, it does not accept it like this. It needs something like `["u"]` instead of `("u",)`